packages/runtime-vapor/src/on.ts

```diff
@@ -1,8 +1,13 @@
+import { getCurrentEffect, onEffectCleanup } from '@vue/reactivity'
+
 export function on(
   el: HTMLElement,
   event: string,
   handler: () => any,
   options?: AddEventListenerOptions,
 ) {
   el.addEventListener(event, handler, options)
+  if (getCurrentEffect()) {
+    onEffectCleanup(() => el.removeEventListener(event, handler, options))
+  }
 }
```

Comment:
The runtime of `v-on` has been changed to be driven by `getCurrentEffect ` and `onEffectCleanup`.